docker-compose.test-ci.yml

```diff
@@ -18,4 +18,15 @@ services:
     image: todo-app:tests-e2e
     command: --ignore todo_app/tests
     volumes:
-      - ./todo_app:/app/todo-app/todo_app
\ No newline at end of file
+      - ./todo_app:/app/todo-app/todo_app
+    depends_on:
+      - mongo-db
+  mongo-db:
+    image: mongo:7.0
+    container_name: mongo_container
+    restart: always
+    ports:
+      - 27017:27017
+    environment:
+    - MONGO_INITDB_ROOT_USERNAME=${MONGO_INITDB_ROOT_USERNAME}
+    - MONGO_INITDB_ROOT_PASSWORD=${MONGO_INITDB_ROOT_PASSWORD}
```

Comment:
Ah so you have already set this up.. is there any reason not to just add these two variables to the `.env.test` file, and pass through that as an `env_file` option?

Reply:
I have add them to the `.env.test` file.  The pipeline has now failed the snyk scan:

![image](https://github.com/CameronB9/DevOps-Course-Starter/assets/135842865/38288c7f-7941-443b-88d8-abaf06286511)

It appears there isn't an upgrade available. What would usually happen in this scenario? Can I disable it for now or maybe allow that it can fail?

Reply:
Hmm, is that error definitely relevant to this change? I feel like that's moaning about a library that we haven't changed here? 🤔

Allowing it to fail and acknowledging that risk might be a reasonable thing (in many places, that would be my default), in some places that would be a nice hard block and we need to go rip the dependency out (potentially a lot more work!)

If I am reading it correctly, allowing it to fail and putting in a ticket to check that we've fixed it in a week or two when we hope the package will have published their fix would be my own approach

Reply:
Sorry, no it's definitely not related to the change I made. As you say it's complaining about one of the existing libraries. I will allow it to fail without stopping the pipeline and check for an update in a few weeks.

Reply:
I have updated the step and set "continue-on-error" to true. 

In the meantime I started work on module 11. Because I didn't merge into main, I created the module 11 branch directly from 10. If I am adding 1 feature at a time, I would merge into main and then create the next branch from main. My current branch is now going to be missing the merge from module 10 into main and it usually complains about commits missing. I would usually pull main into the current branch using something like:

`git pull origin main`

I know there is different ways around this, but what do you think is the best way to do it when you are manging multiple branches that end up out of sync?

Reply:
As you say, various ways, but personally I would:
* Merge this branch as you say
* Pull the latest main on your local machine
* Checkout M11's branch
* Use a rebase (`git rebase main`) to pretend that you started from main in the first place
  * Bonus points if you're comfy using an interactive rebase - but be a little careful if doing this as it does involve re-writing history, so there's a risk of losing work that Git usually protects you from
  * (If you want to mitigate the risk, before you do the rebase, create a new branch from the M11 starting point (e.g. called `placeholder-for-protection`) so that if anything goes wrong, you can always go back to that state)

Reply:
The most recent team I was on always did that; just before you raise your PR, rebase onto main - minor, but it meant that we had a Git tree that looked like everything was cleanly completed in perfected sequence, so it risked mistakes during the rebase, but the Git history was really easy to read

Reply:
Am I correct in thinking that rebase will move the new commits to after the last changes to main? The image below is the current state of my branches:

![image](https://github.com/CameronB9/DevOps-Course-Starter/assets/135842865/d9ca132a-46c1-45ad-bbad-dbe762b80e05)

So in theory, the green line would move up and branch off the last point of the main branch? 

The way I would usually do it would simply apply the missing commits on top of the last changes to the module 11 branch?

Reply:
That's exactly the idea for the rebase, it should just move you to directly on top of main. Whereas merging main into exercise-11 will create a fresh merge commit on top of M11, so they match, but the history is a bit more complex. If you're not sure, create a placeholder branch (or even just make sure not to push, and you can use the origin/exercise-11 branch) and give it a go!

(Bonus - if you think there's any risk, then duplicate the codebase so you've got a copy you don't touch on your machine, and when it all goes fine we can just delete the duplicate 😄 )